plugins/innogrit/innogrit-nvme.c

```diff
@@ -0,0 +1,534 @@
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <errno.h>
+#include <unistd.h>
+#include <sys/stat.h>
+
+#include "common.h"
+#include "nvme.h"
+#include "libnvme.h"
+#include "nvme-print.h"
+#include "typedef.h"
+
+#define CREATE_CMD
+#include "innogrit-nvme.h"
+
+
+/**********************************************************************
+*
+* Entry Point: innogrit_list
+*
+***********************************************************************/
+static int innogrit_list(int argc, char **argv, struct command *command, struct plugin *plugin)
+{
+	int err = 0;
+	int fd;
+	nvme_root_t nvme_root;
+	nvme_host_t h;
+	nvme_subsystem_t s;
+	nvme_ctrl_t c;
+	nvme_ns_t n;
+	struct nvme_id_ctrl ctrl;	
+	const char dash[100] = { [0 ... 99] = '-' };
+
+	printf("%-21s %-20s %-40s %-9s %-26s %-16s %-8s\n", "Node", "SN", "Model", "Namespace", "Usage", "Format", "FW Rev");
+	printf("%-.21s %-.20s %-.40s %-.9s %-.26s %-.16s %-.8s\n", dash, dash, dash, dash, dash, dash, dash);
+
+	nvme_root = nvme_scan(NULL);
+	if (nvme_root) {
+		nvme_for_each_host(nvme_root, h)
+		{
+			nvme_for_each_subsystem(h, s)
+			{
+				nvme_subsystem_for_each_ctrl(s, c)
+				{
+					fd = nvme_ctrl_get_fd(c);
+					nvme_identify_ctrl(fd, &ctrl);
+					if (ctrl.vid == 0x1DBE) 
+					{
+						nvme_ctrl_for_each_ns(c, n) 
+						{
+							nvme_show_list_item(n);
+						}
+					}
+
+				}
+			}
+		}
+
+
+
+	} else {
+		fprintf(stderr, "Failed to scan nvme subsystems\n");
+		err = -errno;
+	}
+
+	nvme_free_tree(nvme_root);
+
+	return err;
+}
+
+
+
+
+
+
+
+/**********************************************************************
+*
+* Entry Point: innogrit_smart_log_additional
```

Comment:
Either at proper function documentation or leave it away. There is no additional information which I can't read from the function definition.